torchao/dtypes/affine_quantized_tensor.py

```diff
@@ -1119,24 +1119,24 @@ def _aqt_is_int8(aqt):
     """Check if an AffineQuantizedTensor is int8 quantized Tensor"""
     return (
         aqt.layout_tensor.dtype == torch.int8 and
-        aqt.quant_min is None or aqt.quant_min == -128 and
-        aqt.quant_max is None or aqt.quant_max == 127
+        (aqt.quant_min is None or aqt.quant_min == -128) and
+        (aqt.quant_max is None or aqt.quant_max == 127)
     )
 
 def _aqt_is_int8_reduced_range(aqt):
     return (
         aqt.layout_tensor.dtype == torch.int8 and
-        aqt.quant_min == -127 and
-        aqt.quant_max is None or aqt.quant_max == 127
+        (aqt.quant_min is not None and aqt.quant_min == -127) and
+        (aqt.quant_max is None or aqt.quant_max == 127)
     )
 
 def _aqt_is_uint4(aqt):
```

Comment:
also probably would be better to change the name to something like `_aqt_is_tensor_core_tiled_uint4` (since this is the format expected by that layout, so users don't confuse this with a "general uint4 check"